api/src/controllers/website.js

```diff
@@ -10,90 +10,44 @@ const { fillErrorObject } = require('../middleware/error');
  * Handle GET request from /clientWebsite/:team_id
  * @param {*} req request object, containing teamId
  * @param {*} res response object
- * @returns 200: the team's website info was found
- * @returns 404: team's website info is not found
+ * @returns 200: return the team's website info
+ * @returns 404: team's website info not found
  * @returns 500: server error
  */
-function getWebPageDetails(req, res) {
-  Website.findOne({
-    teamId: req.params.team_id,
-  })
-    .then((websiteInfo) => {
-      if (!websiteInfo) {
-        return res
-          .status(404)
-          .send(
-            'No Website info found for this Team (This is likely because they have not added any website information yet)',
-          );
-      }
-      return res.status(200).send(websiteInfo);
-    })
-    .catch((error) => res.status(500).json(`Server Error: ${error.message}`));
-}
-
-/**
- * Initializes info about the website.
- * @param info: info to initialize schema with, must include teamId
- * @returns 200: Initial website Info successfully added to DB
- * @returns 400: Team Id given already has a website created
- */
-function createInitialWebsiteInfo(info) {
-  Website.findOne({ teamId: info.teamId })
-    .then((website) => {
-      if (website == null) {
-        // Create Website data
-        Website.create(info)
-          .then((webData) => webData)
-          .catch((err) => {
-            throw err;
-          });
-      }
-    })
-    .catch((err) => {
-      throw err;
-    });
+async function getWebPageDetails(req, res, next) {
+  const { team_id } = req.params;
+  try {
+    const foundWebsiteInfo = await Website.findOne({ teamId: team_id });
+    if (foundWebsiteInfo) {
+      return res.status(200).json(foundWebsiteInfo);
+    }
+    return next(fillErrorObject(404, 'Validation error', ['No webpage detail found with the given team_id']));
+  } catch (err) {
+    return next(fillErrorObject(500, 'Server error', [err.errors]));
+  }
```

Comment:
Ditto, could you document that that this model is associated with a trigger that will automatically create an empty webpage if a team signs up.

Reply:
In the model files would be the best place I reckon, similar to the above.